core/controller/src/main/scala/whisk/core/loadBalancer/InvokerSupervision.scala

```diff
@@ -105,10 +162,66 @@ class InvokerPool(kv: KeyValueStore, invokerDownCallback: String => Unit) extend
                 case _       =>
             }
     }
+
+    /** Receive activeAcks from Invoker. */
+    activeAckConsumer.onMessage((topic, _, _, bytes) => {
+        val raw = new String(bytes, StandardCharsets.UTF_8)
+        CompletionMessage.parse(raw) match {
+            case Success(m: CompletionMessage) => {
+                self ! InvocationFinishedMessage(m.invoker, m.response.response.statusCode <= 2)
+                activeAckCallback(m)
+            }
+            case Failure(t) => logging.error(this, s"failed processing message: $raw with $t")
+        }
+    })
+
+    /** Receive Ping messages from invokers. */
+    pingConsumer.onMessage((topic, _, _, bytes) => {
+        val raw = new String(bytes, StandardCharsets.UTF_8)
+        PingMessage.parse(raw) match {
+            case Success(p: PingMessage) => self ! p
+            case Failure(t)              => logging.error(this, s"failed processing message: $raw with $t")
+        }
+    })
 }
 
 object InvokerPool {
-    def props(kv: KeyValueStore, cb: String => Unit) = Props(new InvokerPool(kv, cb))
+    def props(kv: KeyValueStore,
+              cb: String => Unit, mp: MessageProducer,
+              ackC: MessageConsumer,
+              acb: CompletionMessage => Unit,
+              pc: MessageConsumer) = Props(new InvokerPool(kv, cb, mp, ackC, acb, pc))
+
+    val testAction = ExecManifest.runtimesManifest.resolveDefaultRuntime("nodejs").map { manifest =>
+        new WhiskAction(
+            namespace = EntityPath("whisk.system"),
+            name = EntityName("invokerHealthTestAction"),
+            exec = new CodeExecAsString(manifest, """function main(params) { return params || {}; }""", None))
+    }
+
+    /** Create an test action on startup if it is not there. */
+    def createTestActionForInvokerHealth(db: EntityStore)(implicit logging: Logging, ec: ExecutionContext): Future[WhiskAction] = {
+        implicit val tid = TransactionId.loadbalancer
+        testAction match {
+            case Some(action) =>
+                WhiskAction.get(db, action.docid).recover {
+                    case _: NoDocumentException => WhiskAction.put(db, action)
+                }.flatMap {
```

Comment:
Attention: This will ignore errors on `WhiskAction.get`. What you're after is

```
WhiskAction.get(...).recoverWith {
    case case _: NoDocumentException => WhiskAction.put(db, action)
}.flatMap(_ => action).andThen {
    case Success(_) => logging.info(this, "Testaction for invokerHealth exists now.")
    case Failure(e) => logging.error(this, s"Error on creating testaction for invokerHealth: $e")
}